x-pack/plugins/ml/public/shared_components/single_metric_viewer/single_metric_viewer.tsx

```diff
@@ -190,9 +182,28 @@ const SingleMetricViewerWrapper: FC<SingleMetricViewerPropsWithDeps> = ({
       }
     },
 
-    [setZoom, setSelectedForecastId]
+    [setZoom, setSelectedForecastId, onForecastIdChange]
   );
 
+  const onForecastComplete = (forecastEndTimestamp?: number) => {
+    const { timefilter } = pluginStart.data.query.timefilter;
```

Comment:
When attached to a case there is no time picker, so after running a new forecast or selecting a forecast to view, the forecast data isn't shown as the end time shown in the chart isn't adjusted.

Would it be simpler to not show the 'Forecast' button when inside a case? Then in a follow up, you could allow the user to specify a forecast to be displayed when adding to a case.

Reply:
As discussed, suggest we remove the Forecast button when inside a case, but pass the forecast ID if one is being viewed when adding to a case from the ML Single Metric Viewer page.

Reply:
Changes to hide the Forecast button for cases in https://github.com/elastic/kibana/pull/188791/commits/58f68cc0d65a2170b4a30dd378639acd05b74c27 look good.